### PR TITLE
rename old `Trainer.train_loop` -> `Trainer.fit_loop` 

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #7938 

Resolves a pending todo for renaming the remaining occurrences of the train_loop property which refers to the old training loop construct.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

**Review by kaushikb11 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -227,6 +227,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated the use of `CheckpointConnector.hpc_load()` in favor of `CheckpointConnector.restore()` ([#7652](https://github.com/PyTorchLightning/pytorch-lightning/pull/7652))
 
 
+- Deprecated the `Trainer.train_loop` property in favor of `Trainer.fit_loop` ([#8025](https://github.com/PyTorchLightning/pytorch-lightning/pull/8025))
```
Comment (Borda) [inline CHANGELOG.md:230]: the `train_loop` have not been released, right? so no reason to confuse user

Comment (justusschock) [reply CHANGELOG.md:230]: @Borda we had this property already before the refactor. With the refactor we only rerouted it :)

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by Borda — APPROVED** [review]
